app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/HotspotService.java

```diff
@@ -115,19 +123,62 @@
     return serviceBinder;
   }
 
+  private void setUpHotspotStateReceiver() {
+    hotspotStateReceiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        String action = intent.getAction();
+        if ("android.net.wifi.WIFI_AP_STATE_CHANGED".equals(action)) {
+          int state = intent.getIntExtra(WifiManager.EXTRA_WIFI_STATE, 0);
+
+          if (WifiManager.WIFI_STATE_ENABLED == state % 10) {
+            Log.v(TAG, "Hotspot enabled");
+          } else {
+            initiateServiceDismissal();
+            Log.v(TAG, "Hotspot disabled");
+          }
+        }
+      }
+    };
+
+    IntentFilter intentFilter = new IntentFilter("android.net.wifi.WIFI_AP_STATE_CHANGED");
+    registerReceiver(hotspotStateReceiver, intentFilter);
+  }
+
+  private void initiateServiceDismissal() {
+    if (ServerUtils.isServerStarted) {
+      ServerUtils.isServerStarted = false;
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        hotspotManager.turnOffHotspot();
+      } else {
+        webServerHelper.stopAndroidWebServer();
```

Comment:
it seems like this should be in `callbackAndDismissService` and that that method should be renamed